### PR TITLE
[Security Hardened Kubernetes Cluster] Rule 2003 implementation

### PR DESCRIPTION
**Pull request by georgibaltiev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #356 

**Special notes for your reviewer**:

**Release note**:

```feature user
Implementation for rule `2003` from the `security-hardened-k8s` ruleset for provider `managedk8s`.
```

### CODE REVIEW

**Review by AleksandarSavchev — CHANGES_REQUESTED** [review]
Please update example config with new option.

File: pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2003.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+			}
+			return false, ""
+		}
+	)
+
+	for _, pod := range pods {
+		for _, volume := range pod.Spec.Volumes {
+			volumeTarget := rule.NewTarget("kind", "pod", "name", pod.Name, "namespace", pod.Namespace, "volume", volume.Name)
+			if volume.ConfigMap == nil && volume.CSI == nil && volume.DownwardAPI == nil && volume.EmptyDir == nil &&
+				volume.Ephemeral == nil && volume.PersistentVolumeClaim == nil && volume.Projected == nil && volume.Secret == nil {
```
[the remark this answers is not in the document]
Comment (georgibaltiev) [reply pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2003.go:107]: Should I leave it as it is, or should I reverse the check to look for each of the volume types individually?

[the remark this answers is not in the document]
Comment (AleksandarSavchev) [reply pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2003.go:107]: Let's leave the check as it is. The only improvement we can do is to spread the check in more lines, to be easily readible.

**Review by AleksandarSavchev — APPROVED** [review]
/lgtm

**Pull request merged**